lightning/src/chain/mod.rs

```diff
@@ -189,66 +189,79 @@ pub trait Confirm {
 
 /// An error enum representing a failure to persist a channel monitor update.
 #[derive(Clone, Copy, Debug, PartialEq)]
-pub enum ChannelMonitorUpdateErr {
+pub enum ChannelMonitorUpdateResult {
```

Comment:
In practice, what would that look like? Having a different enum for the status of updating the monitor?

Reply:
My initial take is to make a fourth `Rejected` variant. Then perhaps `ChannelMonitor` could be described in terms of a state machine with enum variants being state transitions? Not sure if that 100% aligns with how things actually work.

Reply:
If its internal only, we can maybe have 4th variant. 
But if we expect clients to understand and return correct value of ChannelMonitorPersistResult, in that case, it might be easier for them if it's just Persist related and separate. (Since it becomes a client-interface kind of thing)
Idk, if it makes things too complicated for us though.

Reply:
Hmm... there's code that uses it for some failure other than persistence (as discussed) and also references `ChannelMonitorUpdateStatus::PermanentFailure` in docs for more general failures, though. So renaming to `ChannelMonitorPersistStatus` would be confusing, IMO. Is the argument for removing `Update` from the name that this is also returned by `watch_channel` and `persist_new_channel`, too?

I'm starting to think that `chain::Watch` and `chainmonitor::Persist` should have their own enum / `Result` types with a `From` conversion, even if there is some overlap. It may make sense to use `Result` to make the conversion easy, which would mean we wouldn't need the `Completed` variant. Then a persist error could map to a single variant in the `chain::Watch` error enum, perhaps wrapping it if we need to treat `InProgess` and `PermanentFailure` differently. Just feels like the idiomatic way of doing this and avoids confusion. The drawback of using `Result` there is `InProgess` is an `Err`, which may be confusing to users?

Reply:
Not sure where we landed on this, but happy to leave any changes to a follow-up.